common/Geometry2d/Rect.cpp

```diff
@@ -18,6 +25,125 @@ bool Rect::intersects(const Rect& other) const {
+            // or BOTTOM), so both must be outside window; exit loop (accept is false)
+            break;
+        } else {
+            // failed both tests, so calculate the line segment to clip
+            // from an outside point to an intersection with clip edge
+            double x, y;
+
+            // At least one endpoint is outside the clip rectangle; pick it.
+            int outcodeOut = outcode0 ? outcode0 : outcode1;
+
+            // Now find the intersection point;
+            // use formulas:
+            //   slope = (y1 - y0) / (x1 - x0)
+            //   x = x0 + (1 / slope) * (ym - y0), where ym is ymin or ymax
+            //   y = y0 + slope * (xm - x0), where xm is xmin or xmax
+            // No need to worry about divide-by-zero because, in each case, the
+            // outcode bit being tested guarantees the denominator is non-zero
+      
+            if (outcodeOut & TOP) {  // point is above the clip window
+                x = x0 + (x1 - x0) * (maxy() - y0) / (y1 - y0);
+                y = maxy();
+            } else if (outcodeOut & BOTTOM) { // point is below the clip window
+                x = x0 + (x1 - x0) * (miny() - y0) / (y1 - y0);
+                y = miny();
+            } else if (outcodeOut & RIGHT) {  // point is to the right of clip window
+                y = y0 + (y1 - y0) * (maxx() - x0) / (x1 - x0);
+                x = maxx();
+            } else if (outcodeOut & LEFT) {   // point is to the left of clip window
+                y = y0 + (y1 - y0) * (minx() - x0) / (x1 - x0);
+                x = minx();
+            }
+            
+            // Now we move outside point to intersection point to clip
+            // and get ready for next pass.
+            if (outcodeOut == outcode0) {
+                x0 = x;
+                y0 = y;
+                Point pt = Point(x0, y0);
+                outcode0 = CohenSutherlandCode(pt);
+                //Save point iff it is inside the Rect
+                if (outcode0==INSIDE){
+                    **nextPoint = pt;
+                    nextPoint = &intr2;
+                }
+            } else {
+                x1 = x;
+                y1 = y;
+                Point pt = Point(x0, y0);
+                **nextPoint = Point(x0, y0);
+                outcode1 = CohenSutherlandCode(pt);
+                //Save point iff it is inside the Rect
+                if (outcode1==INSIDE){
+                    **nextPoint = pt;
+                    nextPoint = &intr2;
+                }
+            }
+        }
+    }
+    return accept;
+}
+
+std::vector<Point> Rect::corners(){
+    std::vector<Point> tmp;
+    tmp.push_back(Point(minx(),miny()));
+    tmp.push_back(Point(minx(),maxy()));
+    tmp.push_back(Point(maxx(),maxy()));
+    tmp.push_back(Point(maxx(),miny()));
+    return tmp;
+}
+
+std::vector<Point> Rect::pointList(){
+    std::vector<Point> tmp;
+    tmp.push_back(pt[0]);
+    tmp.push_back(pt[1]);
+    return tmp;
+}
```

Comment:
I'm not sure if this is a good idea to add. It's already there through the `.pt[0]` and `.pt[1]` values. If it's for debugging and you were trying to print it out or something, it's better to just overload the `<<` operator and do it the correct way

Reply:
That can be a different pr. You don't have to add that

Reply:
removed